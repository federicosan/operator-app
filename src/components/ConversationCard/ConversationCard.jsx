import React, { Component, PropTypes } from 'react';
import moment from 'moment';
import { connect } from 'react-redux';

import { Spinner } from '@blueprintjs/core';

import { openConversation } from '../../store/Chat/actions';

export class ConversationCardComponent extends Component {
  static propTypes = {
    dispatch: PropTypes.func,
    client: PropTypes.objectOf(PropTypes.string),
    chat: PropTypes.string,
    tabIndex: PropTypes.number,
    updateTime: PropTypes.string,
    key: PropTypes.string,
  }

  constructor (props) {
    super(props);

    this.state = {
      client: props.client || false,
      updateTime: props.updateTime,
    };
  }

  render() {
    const { dispatch, chat, tabIndex } = this.props;
    let data = <Spinner className="pt-small" />;

    if (this.state.client) {
      let { name } = this.state.client;

      let relativeTime = moment(this.state.updateTime, 'YYYY-MM-DDTHH:mm:ss.SSS').fromNow();

      data = (
        <button
          tabIndex={tabIndex}
          onClick={() => {
            dispatch(openConversation(chat));
          }}
        >
          <h4>{name}</h4>
          <p>{relativeTime}</p>
        </button>
      );
    }

    return (
      <li id={chat} className="pt-card pt-elevation-1">
        {data}
      </li>
    );
  }
}

const mapStateToProps = state => ({

});

const mapDispatchToProps = dispatch => ({
  dispatch,
});

const ConversationCard = connect(
  mapStateToProps,
  mapDispatchToProps,
)(ConversationCardComponent);

export default ConversationCard;
