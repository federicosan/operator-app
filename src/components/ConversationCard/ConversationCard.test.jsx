import React from 'react';

import { shallow } from 'enzyme';

import ConversationCard from './ConversationCard.jsx';

const store = {
  subscribe: jest.fn(),
  dispatch: jest.fn(),
  getState: jest.fn(() => ({ })),
};

describe('Chat', () => {
  it('matches snapshot', () => {
    const component = shallow(<ConversationCard store={store} />);

    expect(component).toMatchSnapshot();
  });
});
