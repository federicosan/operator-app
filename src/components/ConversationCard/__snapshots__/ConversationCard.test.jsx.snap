exports[`Chat matches snapshot 1`] = `
<ConversationCardComponent
  dispatch={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  } />
`;
